### PR TITLE
Feature/iii 3589 login flow

### PR DESCRIPTION
**Pull request by simon-debruijn**
### Added

-

### Changed

-

### Removed

-

### Fixed

-

---

Ticket: https://jira.uitdatabank.be/browse/III-3589

### CODE REVIEW

**Comment by simon-debruijn** [comment]
@geoffreydhuyvetters Do you see the issue?

File: src/pages/_app.js [hunk 1]
```diff
@@ -44,6 +47,30 @@ Layout.propTypes = {
...
+      if (!cookies?.token) {
+        router.push('/login');
+      }
+    }
+  }, [query, pathname]);
+
+  useEffect(() => {
+    if (user) {
+      setCookie('user', user);
+      // TODO: Currently after logging in it returns to the login page again after loading dashboard, this forces to go back to dashboard
```
Comment (simon-debruijn) [inline src/pages/_app.js:69]: @geoffreydhuyvetters

**Comment by bertramakers** [comment]
Looks good to me aside from the workaround that needs to be fixed, but I'm also not sure what's causing the redirect to `/login` after the `token` is set.

File: src/components/SideBar.js [hunk 2]
```diff
@@ -105,12 +105,21 @@ Menu.propTypes = {
...
       children: t('menu.logout'),
-      onClick: () => {},
+      onClick: () => {
+        removeCookie('token');
+        removeCookie('user');
+
+        const queryString = new URLSearchParams({
+          destination: `${window.location.protocol}//${window.location.host}`,
+        }).toString();
+        window.location.href = `${process.env.NEXT_PUBLIC_AUTH_URL}/logout?${queryString}`;
```
Comment (geoffreydhuyvetters) [inline src/components/SideBar.js:121]: Just out of curiousity. Could we also use router.push to navigate to an external page?

**Review by geoffreydhuyvetters — APPROVED** [review]

File: src/components/SideBar.js (hunk 2 above)
Comment (simon-debruijn) [reply src/components/SideBar.js]: Does seem to work perfectly with router.push. I've changed it

**Pull request merged**